### PR TITLE
Migrate some tests to JUnit 5

### PR DESCRIPTION
**Pull request by asashour**
One hint: `AuthenticationIT` fails locally (even with `4.1` branch), for further investigations.

### CODE REVIEW

File: community/community-it/community-it/src/test/java/org/neo4j/server/http/cypher/integration/QueryResultsSerializationTest.java [hunk 1]
```diff
@@ -106,7 +107,7 @@ public void shouldBeAbleToReturnDeletedEntitiesGraph()
         assertThat( nodesInDatabase() ).isEqualTo( 0L );
     }
 
-    @Test
+    @AfterTest
```
Comment (klaren) [inline community/community-it/community-it/src/test/java/org/neo4j/server/http/cypher/integration/QueryResultsSerializationTest.java]: I think this is a typo?

**Review by klaren — CHANGES_REQUESTED** [review]
Thank you for the PR.

There are a few failing tests in the `org.neo4j.server.` pagacke:

```
        integration.DeadlockIT org.neo4j.server.http.cypher
        TemporalTypeIT.shouldWorkWithLocalTime org.neo4j.server.http.cypher.integration
        TemporalTypeIT.shouldWorkWithLocalDateTime org.neo4j.server.http.cypher.integration
        TemporalTypeIT.shouldWorkWithDate org.neo4j.server.http.cypher.integration
        TemporalTypeIT.shouldWorkWithTime org.neo4j.server.http.cypher.integration
        TemporalTypeIT.shouldWorkWithDateTime org.neo4j.server.http.cypher.integration
        TemporalTypeIT.shouldWorkWithDuration org.neo4j.server.http.cypher.integration
        TemporalTypeIT.shouldOnlyGetNodeTypeInMetaAsNodeProperties org.neo4j.server.http.cypher.integration
        rest.DisableWADLIT org.neo4j.server
        server.NeoWebServerDefaultPortAndHostnameIT org.neo4j
        RowFormatMetaFieldTestIT.metaFieldShouldGetCorrectIndex org.neo4j.server.http.cypher.integration
        integration.TransactionIT org.neo4j.server.http.cypher
        cypher.TransactionTestIT org.neo4j.server.http
        integration.ClientErrorIT org.neo4j.server.http.cypher
        DiscoveryServiceIT.shouldRespondWith200WhenRetrievingDiscoveryDocument org.neo4j.server.rest
        DiscoveryServiceIT.shouldRedirectOnHtmlRequest org.neo4j.server.rest
        DiscoveryServiceIT.shouldGetContentLengthHeaderWhenRetrievingDiscoveryDocument org.neo4j.server.rest
        DiscoveryServiceIT.shouldHaveJsonMediaTypeWhenRetrievingDiscoveryDocumentorg.neo4j.server.rest
        integration.TransactionErrorIT org.neo4j.server.http.cypher
        XForwardFilterIT.shouldPickFirstXForwardedHostHeaderValueFromCommaOrCommaAndSpaceSeparatedList org.neo4j.server.rest.repr
        XForwardFilterIT.shouldUseBaseUriOnBadXForwardedHostHeader org.neo4j.server.rest.repr
        XForwardFilterIT.shouldUseXForwardedHostAndXForwardedProtoHeadersInCypherResponseRepresentations org.neo4j.server.rest.repr
        XForwardFilterIT.shouldUseXForwardedHostAndXForwardedProtoHeadersWhenPresent org.neo4j.server.rest.repr
        XForwardFilterIT.shouldUseXForwardedProtoHeaderWhenPresent org.neo4j.server.rest.repr
        integration.PointTypeIT org.neo4j.server.http.cypher
        server.NeoWebServerIT org.neo4j
        integration.RowFormatMetaFieldTestIT org.neo4j.server.http.cypher
        NeoWebServerDefaultPortAndHostnameIT.shouldDefaultToLocalhostOfNoneSpecifiedInConfig org.neo4j.server
        NeoWebServerDefaultPortAndHostnameIT.shouldDefaultToSensiblePortIfNoneSpecifiedInConfig org.neo4j.server
        DisableWADLIT.should404OnAnyUriEndingInWADL org.neo4j.server.rest
        TransientErrorIT.unavailableCsvResourceShouldRollbackTransaction org.neo4j.server.http.cypher.integration
        TransientErrorIT.deadlockShouldRollbackTransaction org.neo4j.server.http.cypher.integration
```

**Comment by asashour** [comment]
Thanks, I have corrected the `AfterTest`.

Strange, `TemporalTypeIT` passes locally with both IntelliJ and maven.

Could you please send the error log?

**Comment by klaren** [comment]
The problem was that the junit5 tests executes in parallel. We've added the correct annotation to prevent this and rebased the code so it should be merged in a bit.

**Pull request closed without merging**